src/test/features/response/form/models/statement-of-means/supportedByYou.ts

```diff
@@ -0,0 +1,154 @@
+      expect(actual.doYouSupportAnyone).to.be.eq(undefined)
+      expect(actual.numberOfPeople).to.be.eq(undefined)
+    })
+
+    it('should return SupportedByYou with populated only doYouSupportAnyone', () => {
+      const actual = new SupportedByYou().deserialize({ doYouSupportAnyone: false })
+
+      expect(actual.doYouSupportAnyone).to.be.eq(false)
+      expect(actual.numberOfPeople).to.be.eq(undefined)
+    })
+
+    it('should return fully populated SupportedByYou', () => {
+      const actual = new SupportedByYou().deserialize(
+        { doYouSupportAnyone: true, numberOfPeople: { value: 1, details: 'my story' } }
+      )
+
+      expect(actual.doYouSupportAnyone).to.be.eq(true)
+      expect(actual.numberOfPeople.value).to.be.eq(1)
+      expect(actual.numberOfPeople.details).to.be.eq('my story')
+    })
+
+    it('should NOT populate other fields when doYouSupportAnyone = false', () => {
+      const actual = new SupportedByYou().deserialize(
+        { doYouSupportAnyone: false, numberOfPeople: { value: 1, details: 'my story' } }
+      )
+
+      expect(actual.doYouSupportAnyone).to.be.eq(false)
+      expect(actual.numberOfPeople).to.be.eq(undefined)
+    })
+  })
+
+  describe('fromObject', () => {
+
+    it('should return empty SupportedByYou for undefined given as input', () => {
+      const actual = SupportedByYou.fromObject(undefined)
+
+      expect(actual).to.be.eq(undefined)
+    })
+
+    it('should return SupportedByYou with populated only doYouSupportAnyone', () => {
+      const actual = SupportedByYou.fromObject({ doYouSupportAnyone: false })
+
+      expect(actual.doYouSupportAnyone).to.be.eq(false)
+      expect(actual.numberOfPeople).to.be.eq(undefined)
+    })
+
+    it('should return fully populated SupportedByYou', () => {
+      const actual = SupportedByYou.fromObject(
+        { doYouSupportAnyone: true, numberOfPeople: { value: '1', details: 'my story' } }
+      )
+
+      expect(actual.doYouSupportAnyone).to.be.eq(true)
+      expect(actual.numberOfPeople.value).to.be.eq(1)
+      expect(actual.numberOfPeople.details).to.be.eq('my story')
+    })
+
+    it('should NOT populate other fields when doYouSupportAnyone = false', () => {
+      const actual = SupportedByYou.fromObject(
+        { doYouSupportAnyone: false, numberOfPeople: { value: '1', details: 'my story' } }
+      )
+
+      expect(actual.doYouSupportAnyone).to.be.eq(false)
+      expect(actual.numberOfPeople).to.be.eq(undefined)
+    })
+  })
+
+  describe('validation', () => {
+    const validator: Validator = new Validator()
+
+    describe('when doYouSupportAnyone is', () => {
+
+      context('undefined', () => {
+
+        it('should reject', () => {
+          const errors = validator.validateSync(new SupportedByYou())
+
+          expect(errors.length).to.equal(1)
+          expectValidationError(errors, GlobalValidationErrors.YES_NO_REQUIRED)
+        })
+      })
+
+      context('true', () => {
```

Comment:
Test(s) for `false` case are missing.

Reply:
https://github.com/hmcts/cmc-citizen-frontend/pull/236/files#diff-1dbf734b1bb59a7d7abf6d85b8bb0967R145

Reply:
refactored